pygeoapi/api.py

```diff
@@ -569,10 +569,13 @@ def get_collection_items(self, headers, args, dataset, pathinfo=None):
             LOGGER.error(exception)
             return headers_, 500, json.dumps(exception)
 
-        LOGGER.debug('processing property parameters')
-        for k, v in args.items():
-            if k not in reserved_fieldnames and k in p.fields.keys():
-                properties.append((k, v))
+        if self.config['datasets'][dataset]['provider']\
+                .get('allow_property_filtering', False):
```

Comment:
Why do we need `allow_property_filtering`?

Reply:
I may have misinterpreted the comment in [Issue 270](https://github.com/geopython/pygeoapi/issues/279) from @justb4 

> Not sure if standard already provides API convention, but IMHO we should apply a common strategy here as well: Provider should indicate if it can handle property filtering, could be a simple boolean.

Personally, I would be okay removing it. I believe this feature is part of the OGC API specification

Reply:
@tomkralidis  Will give my 2cent, IMHO providers should have a way to control property filtering, since some providers may have problems with or it may consume to many resources.  Maybe  it would be better to implement things in a different way, assuming filtering by default and only if `disable_property_filtering` to disable it 

@justb4  What do you think ??

Reply:
Is the question whether a provider implements filtering or not?  I would think that the provider `.fields()` would relay this (advertising through an OpenAPI document), and keeping it empty essentially achieves the same thing?  Backends should then implement property as they wish (no support, or throw an exception, or process accordingly).  Thoughts?

Reply:
I will not comment on the OpenAPI since I am a bit unfamiliar,and yes the question is : "Whether a provider implements filtering or not" (at least on my mind)

Reply:
My [comment in issue 279](https://github.com/geopython/pygeoapi/issues/279#issuecomment-543371254) expanded (AFAICR):

* a Provider may or may not provide Property filtering, mainly for technical reasons, e.g. a CSV or GeoJSON Provider has no native query/filtering function.
* in those cases the `pygeoapi` core can apply or help with explicit filtering while Provider is building a resultset.

My comment was more directed to avoid every Provider without native property filtering to implement their own. We could e.g. filter during result generation. Probably we don't need a boolean property even, only helper functions the Provider can call.

This was intended to be different from *allowing* property filtering.  This is IMHO more *data collection related*: e.g. some may not allow filtering on any property or only selected properties for very big datasets (millions) or for other reasons. This kind of stuff (complex Filters) has killed many a WFS v1/2, and scared off many users ("you put your database online"!). WFS2 stored queries were a safety net hardly used. This could be a general API capability from 'no filtering' to 'selected fields filtering' to 'any field filtering'. For now I am considering `key=value` filtering. I realize there is a whole range of other filters and -languages like CQL and spatial filters. The OpenAPI spec: https://swagger.io/docs/specification/describing-parameters/#query-parameters . 

(I see now that `pygeoapi` accepts properties as `key=value` in query params but these not seem to be passed to the Provider, unless Provider initializes `self.fields` explicitly. See 

* https://github.com/geopython/pygeoapi/blob/master/pygeoapi/api.py#L574 and
* https://github.com/geopython/pygeoapi/blob/master/pygeoapi/provider/base.py#L52

looks like if Provider cannot populate `self.fields` no filtering can be applied. We should somehow improve here, but can be a nasty problem for Providers without metadata).